cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -104,6 +105,15 @@ func GetStaticPodSpecs(cfg *kubeadmapi.MasterConfiguration, k8sVersion *version.
 		}, mounts.GetVolumes(kubeadmconstants.KubeScheduler)),
 	}
 
+	if cfg.CloudProvider == "openstack" {
```

Comment:
might be worth making `openstack` a [constant](https://github.com/kubernetes/kubernetes/tree/master/cmd/kubeadm/app/constants) and also referencing different providers in the unit tests

Reply:
@jamiehannaford i see `"aws"` and `"gce"` as well. i'll do it in another PR after this one merges

Reply:
okay sgtm

Reply:
I'd rather this be more generic enablement, b/c this isn't openstack specific.  This is a specific os-deployment issue.

Reply:
Should i remove the if condition for openstack?

Reply:
Do AWS and GCE also use the config drive for node metadata?

Reply:
I don't know for sure @jamiehannaford

Reply:
@dims I'd prefer an opt-in knob to enable privileged (default false), b/c again, this is very specific and not necessarily the common case.

Reply:
Agree with a specific opt-in knob.  I'd consider it acceptable to have documentation for that knob state that often times cloud providers need it to be set and name openstack specifically.

Reply:
Agree with a specific opt-in knob.  I'd consider it acceptable to have documentation for that knob state that often times cloud providers need it to be set and name openstack specifically.